### ISSUE TITLE
Tooltip on click should remain open until dismissed

### ISSUE THREAD

**Issue by petermakowski** [canonical-web-and-design/react-components#759]
- Detached tooltip does not remain open until dismissed (by using esc key or clicking outside).
- Tooltip does not close when pressing an `Esc` key

We're using a `detached` tooltip version in `react-components` and it's behaviour matches that one of vanilla in that regard. https://vanillaframework.io/docs/patterns/tooltips#detached

Ideally both variants should behave in exactly the same way as it's just an implementation detail whether it's a detached tooltip or not.

**Comment by bartaz** [comment]
@petermakowski I'm not sure what you mean here exactly.

Is this an issue with React implementation of Tooltip? or with Vanilla one?

In Vanilla CSS framework the distinction between "detached" and standard tooltip is that the standard one is pure CSS, it is only triggered by hover and it can't be controlled with JavaScript (it can't stay open after mouse moves away, it can't be controled via key events). The "detached" variant is for situations when we want to control the tooltip via scripting (therefore this is the only "variant" that is controlled via JS and that's why it's used in react components implementation).

**Comment by petermakowski** [comment]
> @petermakowski I'm not sure what you mean here exactly.
> 
> Is this an issue with React implementation of Tooltip? or with Vanilla one?
> 
> In Vanilla CSS framework the distinction between "detached" and standard tooltip is that the standard one is pure CSS, it is only triggered by hover and it can't be controlled with JavaScript (it can't stay open after mouse moves away, it can't be controled via key events). The "detached" variant is for situations when we want to control the tooltip via scripting (therefore this is the only "variant" that is controlled via JS and that's why it's used in react components implementation).

I just opened another issue in Vanilla https://github.com/canonical-web-and-design/vanilla-framework/issues/4412

**Comment by petermakowski** [comment]
>  The "detached" variant is for situations when we want to control the tooltip via scripting (therefore this is the only "variant" that is controlled via JS and that's why it's used in react components implementation).

Gotcha, let's keep limit the scope of this to the detached variant then.
